File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return -1e9
        cnt,mej = 0,1e9
        for n in nums:
            if cnt == 0:
                cnt+=1
                mej = n
            else:
                if n == mej:
                    cnt+=1
                else:
                    cnt-=1
        return mej
            
        

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cur = nums[0];
        int cnt = 1;
        for (int i = 1;i < nums.size();i++) {
            if (nums[i] == cur) {
                cnt += 1;
            } else {
                cnt -= 1;
            }
            if (cnt == 0) {
                cnt = 1;
                cur = nums[i];
            }
        }
        return cur;
    }
};